pkg/infradb/utils.go

```diff
@@ -0,0 +1,294 @@
+				// for replay
+				tempSubs := vrf.prepareObjectsForReplay(componentName, vrfSubs)
+
+				err = infradb.client.Set(vrf.Name, vrf)
+				if err != nil {
+					return nil, nil, err
+				}
+
+				subsForReplay = append(subsForReplay, tempSubs)
+				objectsToReplay = append(objectsToReplay, vrf)
+			}
+		case "logical-bridge":
+			lbsMap := make(map[string]bool)
+			found, err := infradb.client.Get("lbs", &lbsMap)
+			if err != nil {
+				return nil, nil, err
+			}
+
+			if !found {
+				log.Println("gatherObjectsAndSubsToReplay(): No Logical Bridges have been found")
+				continue
+			}
+			for key := range lbsMap {
+				lb := &LogicalBridge{}
+				found, err := infradb.client.Get(key, lb)
+				if err != nil {
+					return nil, nil, err
+				}
+
+				if !found {
+					return nil, nil, ErrKeyNotFound
+				}
+
+				// tempSubs holds the subscribers list to be contacted for every VRF object each time
+				// for replay
+				tempSubs := lb.prepareObjectsForReplay(componentName, lbSubs)
+
+				err = infradb.client.Set(lb.Name, lb)
+				if err != nil {
+					return nil, nil, err
+				}
+				subsForReplay = append(subsForReplay, tempSubs)
+				objectsToReplay = append(objectsToReplay, lb)
+			}
+		case "svi":
+			svisMap := make(map[string]bool)
+			found, err := infradb.client.Get("svis", &svisMap)
+			if err != nil {
+				return nil, nil, err
+			}
+
+			if !found {
+				log.Println("gatherObjectsAndSubsToReplay(): No SVIs have been found")
+				continue
+			}
+			for key := range svisMap {
+				svi := &Svi{}
+				found, err := infradb.client.Get(key, svi)
+				if err != nil {
+					return nil, nil, err
+				}
+
+				if !found {
+					return nil, nil, ErrKeyNotFound
+				}
+
+				// tempSubs holds the subscribers list to be contacted for every VRF object each time
+				// for replay
+				tempSubs := svi.prepareObjectsForReplay(componentName, sviSubs)
+
+				err = infradb.client.Set(svi.Name, svi)
+				if err != nil {
+					return nil, nil, err
+				}
+				subsForReplay = append(subsForReplay, tempSubs)
+				objectsToReplay = append(objectsToReplay, svi)
+			}
+		case "bp":
+			bpsMap := make(map[string]bool)
+			found, err := infradb.client.Get("bps", &bpsMap)
+			if err != nil {
+				return nil, nil, err
+			}
+
+			if !found {
+				log.Println("gatherObjectsAndSubsToReplay(): No Bridge Ports have been found")
+				continue
+			}
+			for key := range bpsMap {
+				bp := &BridgePort{}
+				found, err := infradb.client.Get(key, bp)
+				if err != nil {
+					return nil, nil, err
+				}
+
+				if !found {
+					return nil, nil, ErrKeyNotFound
+				}
+
+				// tempSubs holds the subscribers list to be contacted for every VRF object each time
+				// for replay
+				tempSubs := bp.prepareObjectsForReplay(componentName, bpSubs)
+
+				err = infradb.client.Set(bp.Name, bp)
+				if err != nil {
+					return nil, nil, err
+				}
+				subsForReplay = append(subsForReplay, tempSubs)
+				objectsToReplay = append(objectsToReplay, bp)
+			}
```

Comment:
again, the code in all cases has similarity. Can we unify it instead of just duplicating?

Reply:
Yes I have tried that and I didn't see any obvious solution so the code doesn't become error prone and unreadable. I will open an issue for this one to revisit in the restructuring effort in the future. Hope that is alright